modem: cmux: fix frame data length encoding

tomi-font opened the pull request:
In cases where the data is bigger than 127 bytes, the first bit of the second byte of the data length field used to always be set.

This is wrong as according to the 3GPP 27.010 spec only the first bit of the first byte is the EA bit; all the others denote the data length.

bjarki-trackunit commented on subsys/modem/modem_cmux.c:
```diff
@@ -228,16 +228,15 @@ static uint16_t modem_cmux_transmit_frame(struct modem_cmux *cmux,
+	if (data_len & ~0x7f) {
 		fcs = crc8(&byte, 1, MODEM_CMUX_FCS_POLYNOMIAL, fcs, true);
 		ring_buf_put(&cmux->transmit_rb, &byte, 1);
-		byte = 0x01 | (data_len >> 7);
-		ring_buf_put(&cmux->transmit_rb, &byte, 1);
+		byte = data_len >> 7;
 	} else {
-		byte = 0x01 | (data_len << 1);
-		ring_buf_put(&cmux->transmit_rb, &byte, 1);
+		byte |= 0x1; /* EA bit to indicate that the length field is one byte long. */
 	}
+	ring_buf_put(&cmux->transmit_rb, &byte, 1);
```
You are correct regarding the error :) However, the changes you are making here are less readable than the code you are replacing (especially `data_len & ~0x7f` instead of `data_len > 127`) :(

You need only change https://github.com/zephyrproject-rtos/zephyr/blob/c05a483ba4251589b713a1231fb6ad4c75cebbaa/subsys/modem/modem_cmux.c#L235C1-L235C33
to
```
byte = data_len >> 7;
```

tomi-font committed:
modem: cmux: fix frame data length encoding

In cases where the data is bigger than 127 bytes,
the first bit of the second byte of the data length field
used to always be set.

This is wrong as according to the 3GPP 27.010 spec
only the first bit of the first byte is the EA bit;
all the others denote the data length.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

bjarki-andreasen reviewed (approved):
Perfect :)
